code/Controllers/CMSMain.php

```diff
@@ -744,8 +746,46 @@ public function getEditForm($id = null, $fields = null)
             $fields->push($liveLinkField = new HiddenField("AbsoluteLink", false, $record->AbsoluteLink()));
             $fields->push($liveLinkField = new HiddenField("LiveLink"));
             $fields->push($stageLinkField = new HiddenField("StageLink"));
+            $fields->push($archiveWarningMsg = new HiddenField("ArchiveWarningMessage"));
             $fields->push(new HiddenField("TreeTitle", false, $record->TreeTitle));
 
+            // Get all page's descendants
+            $record->collateDescendants(true, $descendants);
+            if (!$descendants) {
+                $descendants = [];
+            }
+
+            // Get all campaigns that the page and its descendants belong to
+            $inChangeSetIDs = ChangeSetItem::get_for_object($record)->column('ChangeSetID');
+
+            foreach ($descendants as $page) {
+                $inChangeSetIDs = array_merge($inChangeSetIDs, ChangeSetItem::get_for_object($page)->column('ChangeSetID'));
+            }
+
+            if (count($inChangeSetIDs) > 0) {
+                $inChangeSets = ChangeSet::get()->filter(['ID' => $inChangeSetIDs, 'State' => ChangeSet::STATE_OPEN]);
+            } else {
+                $inChangeSets = new ArrayList();
+            }
+
+            if (count($descendants) > 0 && $inChangeSets->count() > 0) {
+                $archiveWarningTerm = 'CMSMain.ArchiveWarningWithChildrenAndCampaigns';
```

Comment:
Please substitute these two variables with the actually localised value, and invoke `_t(` with the literal key and default values in place. Otherwise i18nTextCollector won't pick these up.